uvicorn/supervisors/multiprocess.py

```diff
@@ -4,24 +4,101 @@
 import os
 import signal
 import threading
-from multiprocessing.context import SpawnProcess
+from multiprocessing import Pipe
 from socket import socket
-from types import FrameType
-from typing import Callable
+from typing import Any, Callable
 
 import click
 
 from uvicorn._subprocess import get_subprocess
 from uvicorn.config import Config
 
-HANDLED_SIGNALS = (
-    signal.SIGINT,  # Unix signal 2. Sent by Ctrl+C.
-    signal.SIGTERM,  # Unix signal 15. Sent by `kill <pid>`.
-)
+UNIX_SIGNALS = {
+    getattr(signal, f"SIG{x}"): x for x in "HUP QUIT TTIN TTOU USR1 USR2 WINCH".split() if hasattr(signal, f"SIG{x}")
+}
 
 logger = logging.getLogger("uvicorn.error")
 
 
+class Process:
+    def __init__(
+        self,
+        config: Config,
+        target: Callable[[list[socket] | None], None],
+        sockets: list[socket],
+    ) -> None:
+        self.real_target = target
+
+        self.parent_conn, self.child_conn = Pipe()
+        self.process = get_subprocess(config, self.target, sockets)
+
+    def ping(self, timeout: float = 5) -> bool:
+        self.parent_conn.send(b"ping")
+        if self.parent_conn.poll(timeout):
+            self.parent_conn.recv()
+            return True
+        return False
+
+    def pong(self) -> None:
+        self.child_conn.recv()
+        self.child_conn.send(b"pong")
+
+    def always_pong(self) -> None:
+        while True:
+            self.pong()
```

Comment:
I would test this out and see what this might do for CPU if the pipe gets flooded. Might need some throttle in there.

Reply:
pong blocks until a packet is received. So there won't be much CPU consumption here.